File: share/pegasus/examples/simple-examples/process.java
import edu.isi.pegasus.planner.dax.*;

public class process {

    public static void main(String[] args) {
        ADAG dax = new ADAG("process");

        File listing = new File("listing.txt");

        Job ls = new Job("ID1", "ls");
        ls.addArgument("-l").addArgument("/");
        ls.setStdout(listing);
        ls.uses(listing, File.LINK.OUTPUT, File.TRANSFER.TRUE, false);
        dax.addJob(ls);

        dax.writeToSTDOUT();
    }
}

File: share/pegasus/examples/simple-examples/process.py
#!/usr/bin/env python3

import sys
from Pegasus.DAX3 import *

dax =ADAG("process")

listing = File("listing.txt")

job = Job("ls")
job.addArguments("-l","/")
job.setStdout(listing)
job.uses(listing, link=Link.OUTPUT, transfer=True, register=False)
dax.addJob(job)

dax.writeXML(sys.stdout)

